axelrod/strategies/meta.py

```diff
@@ -533,3 +535,83 @@ def __init__(self):
                 <= 1]
         super().__init__(team=team)
         self.classifier["long_run_time"] = False
+
+class MemoryDecay(MetaPlayer):
+    """
+    A player utilizes the (default) Tit for Tat strategy for the first (default) 15 turns,
+    at the same time memorizing the opponent's decisions. After the 15 turns have
+    passed, the player calculates a 'net cooperation score' (NCS) for his opponent,
+    weighing decisions to Cooperate as (default) 1, and to Defect as (default)
+    -2. If the opponent's NCS is below 0, the play defects; otherwise,
+    they cooperate.
+
+    The player's memories of the opponent's decisions have a random chance to be
+    altered (i.e., a C decision becomes D or vice versa; default probability
+    is 0.03) or deleted (default probability is 0.1).
+
+    It's necessary to specify EXACT name of the starting strategy if changing
+    the default. Possible strategies can be accessed with the .team attribute.
+
+    Name: Memory Decay
+    """
+    name = 'Memory Decay'
+    classifier = {
+        'memory_depth' : float('inf'),
+        'long_run_time' : False,
+        'stochastic' : True,
+        'makes_use_of' : set(),
+        'inspects_source' : False,
+        'manipulates_source' : False,
+        'manipulates_state' : False
+    }
+
+    def __init__(self, p_memory_delete: float = 0.1, p_memory_alter: float = 0.03,
+                 loss_value: float = -2, gain_value: float = 1,
+                 memory: list = None, start_strategy: Player = TitForTat,
+                 start_strategy_duration: int = 15):
+        super().__init__(team = [start_strategy])
+        self.classifier["stochastic"] = True
+        self.p_memory_delete = p_memory_delete
+        self.p_memory_alter = p_memory_alter
+        self.loss_value = loss_value
+        self.gain_value = gain_value
+        self.memory = [] if memory == None else memory
+        self.start_strategy_duration = start_strategy_duration
+
+    def __repr__(self):
+        return Player.__repr__(self)
+
+    # translates the actions (D and C) to numeric values (loss_value and
+    # gain_value)
+    def gain_loss_tr(self):
+        self.gloss_values = [*map(lambda x: self.loss_value if x == D else
+                                         self.gain_value, self.memory)]
+
+    # alters memory entry, i.e. puts C if there's a D and vice versa
+    def mem_alter(self):
```

Comment:
please use a full name for the method `memory_alter`, `alter_memory` or `flip_memory` and a docstring instead of a comment